doc/custom-widgets.md

```diff
@@ -0,0 +1,139 @@
+## Customize widgets
+
+Angular InstantSearch comes with a set of UI widgets. Some of them have options to modify the HTML output (`<ng-template>`).
```

Comment:
From reading the introduction I do not understand what is the purpose of the guide. Maybe turn it this way:

```txt
Angular InstantSearch comes with a set of UI widgets. Some of them have options to modify the HTML output (`<ng-template>`). But sometime that is not sufficient and you want to completely customise the rendering of a widget.

For this you can use our BaseWidget class.
```